core/src/main/scala/org/apache/spark/scheduler/SparkListener.scala

```diff
@@ -249,6 +250,8 @@ trait SparkListener {
    * Called when other events like SQL-specific events are posted.
    */
   def onOtherEvent(event: SparkListenerEvent) { }
+
+  // WHENEVER WE ADD A METHOD HERE, PLEASE ALSO UPDATE SparkFirehoseListener.
```

Comment:
That's not a bad idea actually -- would require a lot of code changes though ...